.github/workflows/atfe_nightly_build_and_test.yml

```diff
@@ -0,0 +1,64 @@
+# Copyright (c) 2025, Arm Limited and affiliates.
+# Part of the Arm Toolchain project, under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# This workflow runs all build scripts in parallel on Linux.
+# It is intended to be triggered as part of a **nightly build** to
+# validate all build configurations and test them automatically.
+#
+# Each build script is paired with a corresponding test script using
+# a matrix strategy. Each build+test pair runs as a separate job,
+# allowing for concurrency and clear traceability of failures.
+
+name: ATfE Nightly Build and Test
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 2 * * *'  # Every day at 02:00 UTC
+
+jobs:
+  build-and-test-toolchain:
+    name: Build ${{ matrix.build_script }} + Test ${{ matrix.test_script }} on ${{ matrix.runner }}
+    runs-on: ${{ matrix.runner }}
+
+    strategy:
+      fail-fast: false  # Prevents one job failure from cancelling all
+      matrix:
+        include:
+          - build_script: build.sh
+            test_script: test.sh
+            runner: ubuntu-22.04
+          - build_script: build_llvmlibc_overlay.sh
+            test_script: ''
+            runner: ubuntu-22.04
+          - build_script: build_newlib_nano_overlay.sh
+            test_script: ''
+            runner: ubuntu-22.04
+          - build_script: build_newlib_overlay.sh
+            test_script: ''
+            runner: ubuntu-22.04
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Install dependencies
+        run: ./arm-software/embedded/scripts/install_dependencies.sh
+
+      - name: Build ${{ matrix.build_script }}
```

Comment:
Before building, you currently also need to apply the patches from `arm-software/embedded/llvm-project` and `arm-software/embedded/llvm-project-perf` to the checked out repo.
This can be done using the `arm-software/embedded/cmake/patch_repo.py` script.